
# Simulation settings
generic_hw_control_loop:
  loop_hz: 100
  cycle_time_error_threshold: 0.02

# MoveIt-specific simulation settings
moveit_sim_hw_interface:
  joint_model_group: hydra_planning_group
  joint_model_group_pose: home

# Settings for generic ros_control hardware interface
hardware_interface:
  sim_control_mode: 0 # position, 1: velocity
  joints:
    - rotary_table_positioner
    - rob1_joint_1
    - rob1_joint_2
    - rob1_joint_3
    - rob1_joint_4
    - rob1_joint_5
    - rob1_joint_6
    - rob2_joint_1
    - rob2_joint_2
    - rob2_joint_3
    - rob2_joint_4
    - rob2_joint_5
    - rob2_joint_6
    - rob3_joint_1
    - rob3_joint_2
    - rob3_joint_3
    - rob3_joint_4
    - rob3_joint_5
    - rob3_joint_6

# Publish all joint states ----------------------------------
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 100

# Position Joint Trajectory Controller -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
position_trajectory_controller:
   type: position_controllers/JointTrajectoryController
   joints:
    - rotary_table_positioner
    - rob1_joint_1
    - rob1_joint_2
    - rob1_joint_3
    - rob1_joint_4
    - rob1_joint_5
    - rob1_joint_6
    - rob2_joint_1
    - rob2_joint_2
    - rob2_joint_3
    - rob2_joint_4
    - rob2_joint_5
    - rob2_joint_6
    - rob3_joint_1
    - rob3_joint_2
    - rob3_joint_3
    - rob3_joint_4
    - rob3_joint_5
    - rob3_joint_6
   constraints:
      goal_time: 1.0 # If the timestamp of the goal trajectory point is t, then following the trajectory succeeds if it reaches the goal within t +/- goal_time, and aborts otherwise.
      stopped_velocity_tolerance: 0.01 # Velocity to be considered approximately equal to zero
